### ISSUE TITLE
A-pose garments?

### ISSUE THREAD

**Issue by ligengen** [Dolorousrtur/HOOD#6]
Hi, thanks for the work! I am wondering if your work can be used for other datasets, e.g. BEDLAM, where the rest poses of all clothes are in A-pose. I tried to modify [smplx_v_rest_pose](https://github.com/Dolorousrtur/HOOD/blob/3b3392ea315cdf76a49c10c212a7df0e25e793bc/utils/mesh_creation.py#L82) when adding garments into dict but without luck (but it seems that the upper part looks alright): Anything else I need to change? Thanks a lot!

<img width="258" alt="Screenshot 2023-08-25 at 03 46 29" src="https://github.com/Dolorousrtur/HOOD/assets/26115538/70f48d67-c90d-42e4-96bd-cca46f0d9cf3">

**Comment by Dolorousrtur** [comment]
The code initializes the garment geometry for the first 2 frames with LBS using the canonical SMPL body.
In your case, you would need to initialize the garment geometry from an existing mesh in the A-pose and not rely on LBS.

I hope to add the functionality to start inference from arbitrary poses without LBS next week if I have time.

**Comment by ligengen** [comment]
Hi! Thanks for the reply! But I did not quite get you. All of the garments are in A-rest pose and I did change that line to `smplx_v_rest_pose = smpl_model(body_pose=A_POSE).vertices[0].detach().cpu().numpy()`. Could you please refer to the exact code lines that I need to change? Thanks a lot!

**Comment by ligengen** [comment]
To whom it may concern: you need to multiply an inverse matrix (zero-pose to A-pose transformation) [here](https://github.com/Dolorousrtur/HOOD/blob/e2c92553945890af7b3f30eec4c2614b2822c1eb/utils/lbs.py#L81) and also use correct [pose_offsets](https://github.com/Dolorousrtur/HOOD/blob/e2c92553945890af7b3f30eec4c2614b2822c1eb/utils/lbs.py#L97)

**Comment by JPPPPPF** [comment]
Hello @ligengen , would you mind sharing the code changes you did please?
